pkg/controllers/syncer/syncer.go

```diff
@@ -105,14 +106,23 @@ func (r *syncerController) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		if !kerrors.IsNotFound(err) {
 			return ctrl.Result{}, err
 		}
-
 		pObj = nil
 	}
 
 	// check if we should skip resource
-	// this is to distinguish generic and plugin syncers with the core syncers
-	if pObj != nil && r.excludePhysical(pObj) {
-		return ctrl.Result{}, nil
+	if pObj != nil {
+		// check if physical object is actually managed by vcluster or just coincidentally named
+		isManaged, err := r.syncer.IsManaged(ctx, pObj)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to check if physical object is managed: %w", err)
+		}
+		if !isManaged {
```

Comment:
This should be an `else if !isManaged {`

Reply:
@rohantmp yes but this is semantically still part of the above statement and looks cleaner

Reply:
@rohantmp any updates here?